engine/src/main/java/org/hibernate/validator/internal/constraintvalidators/bv/number/bound/NumberComparatorHelper.java

```diff
@@ -41,14 +50,14 @@ public static int compare(Double number, long value, OptionalInt treatNanAs) {
 		if ( infinity.isPresent() ) {
 			return infinity.getAsInt();
 		}
-		return Long.compare( number.longValue(), value );
+		return Double.compare( number, value );
 	}
 
 	public static int compare(Float number, long value, OptionalInt treatNanAs) {
 		OptionalInt infinity = InfinityNumberComparatorHelper.infinityCheck( number, treatNanAs );
 		if ( infinity.isPresent() ) {
 			return infinity.getAsInt();
 		}
-		return Long.compare( number.longValue(), value );
+		return Double.compare( number, value );
```

Comment:
Is there a reason why you don't use `Float#compare()` here?

Reply:
It was too early in the morning and I thought that `Long` wouldn't "fit" in float :smile: